adapters/repos/db/lsmkv/store.go

```diff
@@ -446,6 +447,10 @@ func (s *Store) ReplaceBuckets(ctx context.Context, bucketName, replacementBucke
 	currReplacementBucketDir := replacementBucket.dir
 	newReplacementBucketDir := currBucketDir
 
+	if err := bucket.Shutdown(ctx); err != nil {
```

Comment:
shutting down the bucket to be replaced at this point seems asserted.

I think we need to prevent flushing and compaction on the replacementBucket before renaming the paths.  Just saw that `updateBucketDir` is acquiring the required locks but I think it's not enough, the flushing and maintenance locks may be acquired before doing the path renaming.

Reply:
also renaming the path in the in-memory data struct for the bucket being replaced may not be done since that bucket is already shut down

Reply:
ah, that must be why it was happening afterwards, in the original code.  It does seem to work, but I can try and get the locks instead